Upgrade to driver version 2.0.0-alpha02

henriknyman opened the pull request:
- Read the actual database name in the ResultSummary from server
- Do not set database session parameter if absent

henriknyman committed:
Upgrade to driver version 2.0.0-alpha02

- Read the actual database name in the ResultSummary from server
- Do not set database session parameter if absent

henriknyman committed:
Adapt to rename of Order field in plan description

henriknyman committed:
Display connection failure in the prompt

- Handle non-existing database name in ResultSummary

henriknyman committed:
Clean up database connection state handling

- Separate state handling from interactive prompt display
- Refresh on new results from server

sherfert commented on cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java:
```diff
@@ -161,16 +157,18 @@ private void reconnect(boolean keepBookmark) {
 
         String query = activeDatabaseNameAsSetByUser.equals(SYSTEM_DB_NAME) ? "SHOW DATABASES" : "RETURN 1";
+
+        resetActualDbName(); // Set this to null first in case run throws an exception
```
I think it would be nice to capture this behavior in a test.

sherfert reviewed (commented):
There are some test failures, and the last commit coul dneed some additional test coverage,

henriknyman committed:
Fix BoltStateHandlerTests

henriknyman committed:
Add tests for actual database name

henriknyman committed:
Distinguish between DatabaseNotFound and DatabaseUnavailable

- Show different error status in the prompt based on the error code
- Color the error status red in the prompt
- Do not reset actual name before running user query

henriknyman committed:
Clean up error prompt handling

- Move tracking of last error code into StatementExecuter
- Remove the default unknown error prompt option

henriknyman commented:
@sherfert I added a separate card to add some integration tests snooping on how the error state shows in the prompt while starting and stopping databases etc. We would probably need to make the `ConsoleReader` an injectable dependency.
